cmd/update.go

```diff
@@ -4,40 +4,220 @@
 package cmd
 
 import (
+	"context"
+	"crypto/sha256"
 	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
 
 	"github.com/alexellis/arkade/pkg"
+	"github.com/alexellis/arkade/pkg/env"
+	"github.com/alexellis/arkade/pkg/get"
+	"github.com/alexellis/go-execute/v2"
 	"github.com/morikuni/aec"
 	"github.com/spf13/cobra"
 )
 
 func MakeUpdate() *cobra.Command {
 	var command = &cobra.Command{
-		Use:          "update",
-		Short:        "Print update instructions",
-		Example:      `  arkade update`,
-		Aliases:      []string{"u"},
-		SilenceUsage: false,
+		Use:   "update",
+		Short: "Replace the running binary with an updated version",
+		Long: `The latest release version of arkade will be downloaded from GitHub.
+
+If that text is not found by running "arkade version", then the release
+will be downloaded, along with its .sha256 checksum file.
+
+If the checksum matches the downloaded file then the running binary will be
+replaced with the new binary.
+
+This command can be run as often as you require, won't download the same 
+version twice.`,
+		Example:       `  arkade update`,
+		Aliases:       []string{"u"},
+		SilenceUsage:  true,
+		SilenceErrors: false,
 	}
-	command.Run = func(cmd *cobra.Command, args []string) {
-		fmt.Println(arkadeUpdate)
 
-		fmt.Println("\n", aec.Bold.Apply(pkg.SupportMessageShort))
+	command.Flags().Bool("verify", true, "Verify the checksum of the downloaded binary")
+	command.Flags().Bool("force", false, "Force a download of the latest binary, even if up to date, the --verify flag still applies")
+
+	command.RunE = func(cmd *cobra.Command, args []string) error {
+
+		verifyDigest, _ := cmd.Flags().GetBool("verify")
+		forceDownload, _ := cmd.Flags().GetBool("force")
+
+		name := "arkade"
+		toolList := get.MakeTools()
+		var tool *get.Tool
+		for _, t := range toolList {
+			if t.Name == name {
+				tool = &t
+				break
+			}
+		}
+
+		release, err := get.FindGitHubRelease("alexellis", name)
+		if err != nil {
+			return err
+		}
+
+		executable, err := os.Executable()
```

Comment:
Tested this on my Intel Mac and found that if a symlink is used the update will replace the symlink instead of the binary it points to.

Reply:
"if a symlink is used" where? If `/usr/local/bin/arkade` is a symlink?

Reply:
Yes, if the path to the arkade binary that is being executed is a symlink.